### ISSUE TITLE
`./configure --enable-llvm` creates fully parallel build, can't be disabled

### ISSUE THREAD

**Issue by jwaldmann** [mono/mono#20388]
## Steps to Reproduce

build from source ( mono-6.12.0.90.tar.xz ) on Fedora GNU/Linux x86_64 (clang version 10.0.0, gcc (GCC) 10.2.1 20200723)
```
./configure --prefix=/opt/mono/mono-6.12.0.90 --disable-crash-reporting --enable-llvm
make -j1
```
This creates a huge process tree. Note: `make -j1` at the top, but `make -j` down below
```
ps a  --forest --width=80
    PID TTY      STAT   TIME COMMAND
   3457 pts/1    Ss     0:00 bash
  68026 pts/1    R+     0:00  \_ ps a --forest --width=80
   3403 pts/0    Ss     0:00 bash
  66816 pts/0    S+     0:00  \_ make -j1
  66819 pts/0    S+     0:00      \_ make all-recursive
  66820 pts/0    S+     0:00          \_ /bin/sh -c fail=; \ if (target_option=k
  66846 pts/0    S+     0:00              \_ /bin/sh -c fail=; \ if (target_opti
  66847 pts/0    S+     0:00                  \_ make all
  66851 pts/0    S+     0:00                      \_ make -j -C /home/me/s
  66854 pts/0    S+     0:00                          \_ make -s -f CMakeFiles/M
  66871 pts/0    S+     0:00                              \_ make -s -f lib/Dema
  66889 pts/0    S+     0:00                              |   \_ /usr/lib64/ccac
  67217 pts/0    S+     0:00                              |       \_ /usr/bin/c+
  67230 pts/0    R+     0:00                              |           \_ /usr/li
  66875 pts/0    S+     0:00                              \_ make -s -f lib/Tabl
  66894 pts/0    S+     0:00                              |   \_ /usr/lib64/ccac
  67103 pts/0    S+     0:00                              |   |   \_ /usr/bin/c+
  67116 pts/0    R+     0:00                              |   |       \_ /usr/li
...
```
This renders my machine fully unresponsive (kernel is busy swapping, I guess).

Without `--enable-llvm`, it's looking normal 
```
    PID TTY      STAT   TIME COMMAND
   3457 pts/1    Ss     0:00 bash
  83856 pts/1    R+     0:00  \_ ps a --forest --width=80
   3403 pts/0    Ss     0:00 bash
  82483 pts/0    S+     0:00  \_ make -j1
  82486 pts/0    S+     0:00      \_ make all-recursive
  82487 pts/0    S+     0:00          \_ /bin/sh -c fail=; \ if (target_option=k
  82518 pts/0    S+     0:00              \_ /bin/sh -c fail=; \ if (target_opti
  82519 pts/0    S+     0:00                  \_ make all
  82520 pts/0    S+     0:00                      \_ /bin/sh -c fail=; \ if (tar
  83754 pts/0    S+     0:00                          \_ /bin/sh -c fail=; \ if 
  83755 pts/0    S+     0:00                              \_ make all
  83842 pts/0    S+     0:00                                  \_ /bin/sh -c echo
  83844 pts/0    S+     0:00                                      \_ /usr/bin/ba
  83851 pts/0    S+     0:00                                          \_ gcc -DH
  83854 pts/0    S+     0:00                                              \_ /us
  83855 pts/0    R+     0:00                                                 /us
   1536 tty1     Ssl+   1:45 /usr/libexec/Xorg -core -noreset :0 -seat seat0 -au
```

**Comment by lambdageek** [comment]
The build parallelism is set in `build.mk` here

https://github.com/mono/mono/blob/ec3bd48aeb44cf1188c1b733218a0ee14efbe825/llvm/build.mk#L48

and `CPU_COUNT` is set here

https://github.com/mono/mono/blob/ec3bd48aeb44cf1188c1b733218a0ee14efbe825/llvm/build.mk#L12

@jwaldmann what does
```sh
/bin/sh -c 'echo -n "count " ; getconf _NPROCESSORS_ONLN ; echo exit code $?'
```

print on your machine.  I bet it's
```
count  
exit code 0
``` 

(on my machine it prints `count 4` and `exit code 0`)

**Comment by jwaldmann** [comment]
```
count 8
exit code 0
```

**Comment by lambdageek** [comment]
huh. well there goes that theory.

Do you have `ninja` installed?  I think you could try installing that as a possible workaround - the LLVM cmake build will use ninja instead of make as the backend and you can sidestep this `-j` issue.  (You might need to do some cleaning of mono's `llvm/build` dir)

**Comment by jwaldmann** [comment]
same thing? (many processes below `ninja`)
```
   3457 pts/1    Ss     0:00 bash
 151284 pts/1    S+     0:00  \_ make -j1
 151287 pts/1    S+     0:00      \_ make all-recursive
 151288 pts/1    S+     0:00          \_ /bin/sh -c fail=; \ if (target_option=k
 151314 pts/1    S+     0:00              \_ /bin/sh -c fail=; \ if (target_opti
 151315 pts/1    S+     0:00                  \_ make all
 151319 pts/1    S+     0:00                      \_ /usr/bin/ninja -C /home/wal
 151320 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151367 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151369 pts/1    R      0:00                          |       \_ /usr/libexec/gc
 151321 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151358 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151360 pts/1    R      0:00                          |       \_ /usr/libexec/gc
 151322 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151356 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151357 pts/1    R      0:01                          |       \_ /usr/libexec/gc
 151323 pts/1    R      0:00                          \_ /usr/lib64/ccache/c++ -
 151326 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151364 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151365 pts/1    R      0:01                          |       \_ /usr/libexec/gc
 151328 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151359 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151361 pts/1    R      0:00                          |       \_ /usr/libexec/gc
 151332 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151366 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151368 pts/1    R      0:01                          |       \_ /usr/libexec/gc
 151371 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151372 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151373 pts/1    R      0:00                          |       \_ /usr/libexec/gc
 151376 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151377 pts/1    S      0:00                          |   \_ /usr/bin/c++ -fPIC 
 151378 pts/1    R      0:00                          |       \_ /usr/libexec/gc
 151380 pts/1    S      0:00                          \_ /usr/lib64/ccache/c++ -
 151381 pts/1    S      0:00                              \_ /usr/bin/c++ -fPIC 
 151382 pts/1    R      0:00                                  \_ /usr/libexec/gc
```

**Comment by lambdageek** [comment]
Hm.  Too bad about ninja.

I just realized however that the CPU_COUNT stuff is only in `master` mono, not in 6.12.  We never backported https://github.com/mono/mono/pull/19798

**Comment by lambdageek** [comment]
The backport PR is here https://github.com/mono/mono/pull/20389

**Comment by jwaldmann** [comment]
> ninja limits itself to NCPUS+2

That figures. I was observing 10 jobs for my 8 cores.

Thanks for the quick reaction, and fix! In what release might this appear, then?

**Comment by lambdageek** [comment]
Once the PR is merged, it'll be in a 6.12.0.* build in (usually) a couple of days.

**Comment by lambdageek** [comment]
This is now fixed in Mono 6.12.0.99 or later
